pkg/provider/kubernetes/ingress/kubernetes.go

```diff
@@ -844,6 +846,32 @@ func makeRouterKeyWithHash(key, rule string) (string, error) {
 	return dupKey, nil
 }
 
+func buildRule(strictPrefixMatching bool, matcher string, path string) string {
+	// When enabled, strictPrefixMatching ensures that prefix matching follows
+	// the Kubernetes Ingress spec (path-element-wise instead of character-wise).
+	if strictPrefixMatching && matcher == "PathPrefix" {
+		// To mimic Kubernetes behavior (see:
+		// https://kubernetes.io/docs/concepts/services-networking/ingress/#examples),
+		// "/v12" should not match "/v1".
+		//
+		// Traefik's default PathPrefix matcher performs a character-wise prefix match,
+		// unlike Kubernetes which matches path elements.
+		// Use PathRegexp to replicate element-wise behavior.
+		//
+		// See TestPrefixMatchRegex() for examples.
+		return "PathRegexp(`" + buildPrefixMatchRegex(path) + "`)"
```

Comment:
I think it would be great using a matcher combination to be consistent with the Kubernetes Gateway provider (which also complies with the Ingress matching specification), something like: https://github.com/traefik/traefik/blob/master/pkg/provider/kubernetes/gateway/httproute.go#L645

Reply:
I think I can extract a common function (in gateway pkg) to build Path Prefix rules to avoid duplicating code. WDYT?